src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2514,13 +2514,25 @@ namespace OpenRCT2::Ui::Windows
 
         void ShowSpecialTrackDropdown(Widget* widget)
         {
+            auto& elements = _specialElementDropdownState.Elements;
+
             int32_t defaultIndex = -1;
-            for (size_t i = 0; i < _specialElementDropdownState.Elements.size(); i++)
+            int32_t i = 0;
+            for (auto& element : elements)
             {
-                OpenRCT2::TrackElemType trackPiece = _specialElementDropdownState.Elements[i].TrackType;
+                TrackElemType trackPiece = element.TrackType;
+
+                // Separate elements logically
+                if (trackPiece == TrackElemType::None)
+                {
+                    gDropdownItems[i++].Format = kStringIdEmpty;
+                    continue;
+                }
 
                 const auto& ted = GetTrackElementDescriptor(trackPiece);
                 StringId trackPieceStringId = ted.description;
+
+                // TODO: this should probably be done elsewhere
```

Comment:
Should it? And where?

Reply:
When we refactor dropdowns, we could add a 'value' property to each of the dropdown items. In that case, we could prepare a (stringId, value) instead of sending the actual track element information for this function to deal with. It would move this logic to the `Construction.cpp` unit.

In the mean time, I don't think there is a better place _yet_. It just seemed strange to have an exception for this one particular track piece right here.

Reply:
It will.